packages/blade/src/components/Button/Button/Button.tsx

```diff
@@ -43,6 +43,32 @@ type ButtonCommonProps = {
    * @private
    */
   'aria-describedby'?: string;
+  /**
+   * It is exposed for internal usage with menu.
+   *
+   * @private
+   */
+  'aria-controls'?: string;
+  /**
+   * It is exposed for internal usage with menu.
+   *
+   * @private
+   */
+  'aria-expanded'?: string;
+  /**
+   * It is exposed for internal usage with menu.
+   *
+   * @private
+   */
+  'aria-haspopup'?: 'menu';
+  /**
+   * It is exposed for internal usage with menu.
+   *
+   * @private
+   */
+  role?: 'menuitem';
+  tabIndex?: BaseButtonProps['tabIndex'];
```

Comment:
Our button will get very polluted with all these props. Lets just use BaseButton for internal usage?

Reply:
Its not really internal usage. The consumers will import and add Button on their end

Reply:
They don't have to explicitly these props because we clone component and pass them internally. But the component that consumers add is still Button itself